kms/api/CloudKmsTest/CloudKmsTest.cs

```diff
@@ -279,6 +294,124 @@ private bool AssertRoleAndMemberPresent(string response, string role, string mem
             }
             return (foundRole && foundMember);
         }
+
+        private CryptoKeyVersionName BuildAsymmetricKey(KeyRingName keyRingName, String cryptoKey,
+                CryptoKey.Types.CryptoKeyPurpose purpose,
+                CryptoKeyVersion.Types.CryptoKeyVersionAlgorithm algorithm)
+        {
+            KeyManagementServiceClient client = KeyManagementServiceClient.Create();
+            CryptoKey cryptoKeyToCreate = new CryptoKey
+            {
+                Purpose = purpose,
+                VersionTemplate = new CryptoKeyVersionTemplate
+                {
+                    Algorithm = algorithm,
+                    ProtectionLevel = ProtectionLevel.Software,
+                },
+            };
+            CryptoKey key = client.CreateCryptoKey(keyRingName, cryptoKey, cryptoKeyToCreate);
+
+            CryptoKeyVersionName ckvName = CryptoKeyVersionName.Parse(key.Name + "/cryptoKeyVersions/1");
+            CryptoKeyVersion ckv = client.GetCryptoKeyVersion(ckvName);
+
+            // The version is generated asynchronously. Wait for it to be in state Enabled before

```

Comment:
When is the version created, when you create the key or when you first ask for it?
I wonder if any of these operations should be long running operations.

Reply:
They aren't modeled as LROs in the API, so this would require handwritten code

Reply:
I'll ask around to see if they should be long running in the API but agree that if not, this is fine as it is.